src/core/coap/coap.cpp

```diff
@@ -95,6 +158,8 @@ ThreadError Client::SendMessage(Message &aMessage, const Ip6::MessageInfo &aMess
 
     SuccessOrExit(error = header.FromMessage(aMessage, 0));
 
+    mResponsesQueue.EnqueueResponse(aMessage, aMessageInfo);
```

Comment:
It's filtered inside `EnqueueResponse`. IMO, it might be better to filter here, which is more clear.